### PR TITLE
Remove dependency on internal VS utility apis.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
These appear to all have been added accidentally.  The APIs are also removed in later VS package releases.

### CODE REVIEW

**Review by dibarbet — APPROVED** [review]
is it possible to 'ban' these methods?  In particular I've hit the GetOrDefault vs GetValueOrDefault case a couple times.

**Review by mavasani — APPROVED** [review]

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
> is it possible to 'ban' these methods?

We can ban the `Microsoft.VisualStudio.Utilities.Internal` namespace.

File: src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs [hunk 1]
```diff
@@ -558,7 +557,7 @@ internal async Task PublishDiagnosticsAsync(IDiagnosticService diagnosticService
                 {
                     // There were diagnostics from other documents, but none from the current document.
                     // If we're tracking the current document, we can stop.
-                    _publishedFileToDiagnostics.GetOrDefault(fileUri)?.Remove(document.Id);
+                    ((IReadOnlyDictionary<Uri, Dictionary<DocumentId, ImmutableArray<LanguageServer.Protocol.Diagnostic>>>)_publishedFileToDiagnostics).GetValueOrDefault(fileUri)?.Remove(document.Id);
```
Comment (genlu) [inline src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs:560]: nit: I think `IReadOnlyDictionaryExtensions.GetValueOrDefault(_publishedFileToDiagnostics, fileUri)` looks better

Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs:560]: Hmm -- I think we have another overload that takes a defaultValue parameter that might make it easier to deal with than this.

**Review by jasonmalinowski — APPROVED** [review]

**Review by genlu — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs]: done.

File: src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs [hunk 2]
```diff
@@ -558,8 +557,7 @@ internal async Task PublishDiagnosticsAsync(IDiagnosticService diagnosticService
                 {
                     // There were diagnostics from other documents, but none from the current document.
                     // If we're tracking the current document, we can stop.
-                    _publishedFileToDiagnostics.GetOrDefault(fileUri)?.Remove(document.Id);
-                    _documentsToPublishedUris.MultiRemove(document.Id, fileUri);
+                    IReadOnlyDictionaryExtensions.GetValueOrDefault(_publishedFileToDiagnostics, fileUri)?.Remove(document.Id);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs:560]:
```suggestion
                    IReadOnlyDictionaryExtensions.GetValueOrDefault(_publishedFileToDiagnostics, fileUri)?.Remove(document.Id);
                    _documentsToPublishedUris.MultiRemove(document.Id, fileUri);
```

**Pull request merged**